components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -217,6 +220,20 @@ private void handleUserProvisioning(String username, UserStoreManager userStoreM
         Map<String, String> userClaims = prepareClaimMappings(attributes);
 
         if (userStoreManager.isExistingUser(username)) {
+            String associatedUserName = FrameworkUtils.getFederatedAssociationManager()
+                    .getUserForFederatedAssociation(tenantDomain, idp, subjectVal);
+
+            if (StringUtils.isBlank(associatedUserName)) {
+                // If a local user is using the same username, association is not allowed unless enabled through config.
+                if (isAssociationToExistingUserAllowed()) {
```

Comment:
Dont' we need to handle migration for this configuration?

Reply:
For asgardeo, it does not associate JIT provisioning users to local accounts, rather creates new accounts. Therefore the default behaviour of this change will not affect asgardeo deployment. 

For other customers whose behaviour is to associate to local users, they will have to activate this through the configuration.